parser.go

```diff
@@ -1103,6 +1113,61 @@ func (p *parser) registerType(pkgPath, pkgName, typeName string) (string, error)
 	return registerTypeName, nil
 }
 
+func trimSplit(csl string) []string {
+	s := strings.Split(csl, ",")
+	for i := range s {
+		s[i] = strings.TrimSpace(s[i])
+	}
+	return s
+}
+
+func (p *parser) handleCompositeType(pkgPath, pkgName, typeName string) (*SchemaObject, error) {
+	re := regexp.MustCompile("(oneOf|anyOf|allOf|not)\\(([^\\)]*)\\)")
+	matches := re.FindStringSubmatch(typeName)
+	if len(matches) < 3 {
+		return nil, nil
+	}
+	op := matches[1]
+	if matches[2] == "" {
+		return nil, fmt.Errorf("Expected 1 or more arguments, received '%s'", typeName)
+	}
+	args := trimSplit(matches[2])
+
+	// not only supports one arg
+	if op == "not" && len(args) != 1 {
+		return nil, fmt.Errorf("Invalid number of arguments for not compound type, expected 1 received %d", len(args))
+	}
+
+	var sobs []*SchemaObject
+	for i := range args {
+		result, err := p.parseBodyType(pkgPath, pkgName, args[i])
+		if err != nil {
+			return nil, err
+		}
+		sobs = append(sobs, result)
+	}
+
+	if op == "not" {
```

Comment:
Can this be made into a switch case statement? https://tour.golang.org/flowcontrol/9

Reply:
Updated

Reply:
And DRY'd it up a bit while I was at it ^_^